api/app/src/main/kotlin/packit/clients/GithubUserClient.kt

```diff
@@ -0,0 +1,86 @@
+package packit.clients
+
+import org.kohsuke.github.*
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import packit.AppConfig
+import packit.exceptions.PackitAuthenticationException
+import packit.model.User
+import packit.security.Role
+import packit.security.profile.UserPrincipal
+
+@Component
+class GithubUserClient(private val config: AppConfig, private val githubBuilder: GitHubBuilder = GitHubBuilder()) {
+
+    private var github: GitHub? = null
+    private var ghUser: GHMyself? = null
+    private val allowedOrgs = config.authGithubAPIOrgs.split(",")
+
+    fun authenticate(token: String)
+    {
+        connectToClient(token)
+        ghUser = getGitHubUser()
+    }
+
+    fun getUser(): UserPrincipal
+    {
+        checkAuthenticated()
+        val ghu = ghUser!!
+
+        val user = User(
+            1L,
+            ghu.login,
+            "",
+            Role.USER,
+            ghu.name ?: "",
+            mutableMapOf()
```

Comment:
We're going to simplify the user class in a future ticket - should just need a string id from the provider, not a numeric id or any further metadata like name.